scripts/msg.h.template

```diff
@@ -64,6 +75,9 @@ struct @(spec.short_name)_
   typedef @(spec.short_name)_<ContainerAllocator> Type;
 
 @# constructors (with and without allocator)
+@[if have_plugin]@
+#ifndef @(spec.package.upper())_MESSAGE_@(spec.short_name.upper())_CUSTOM_CONSTRUCTOR
```

Comment:
The use case of initializing the fields with different values would imo be better addressed by specifying those default values in the `.msg` file (as it is possible in ROS 2). That approach ensures a consistent initialization across all languages. Therefore I would rather not have a `CUSTOM_CONSTRUCTOR` macro.

Or do you have other use cases in mind beside that?

Reply:
I agree that specifying default values in the msg file would be preferable in most cases. But this feature is not available in ROS 1 to my best knowledge and custom constructors could go further, e.g. initialize values nested inside a message field. An example use case would be quaternions nested inside messages that represent orientation and rotation (like in `geometry_msgs/Transform`) which could be initialized to a unit quaternion in that context, while this is not necessarily true for arbitrary quaternions.

Reply:
(just my 2 cents on this, in the hypothesis that we had default values in ROS1)
If it was possible, I do believe that users should be able specify default values for nested fields as for any other field type and generate a constructor accordingly. (so in your example `geometry_msgs/Transform.msg` would specify the default value of the `rotation` field to be a unit Quaternion and the constructor of the `Transform` message in every language would initialize the `rotation` field accordingly)